transform/transformers/MWSSTransformer.py

```diff
@@ -0,0 +1,437 @@
+from decimal import Decimal, InvalidOperation
+from collections import namedtuple
+from collections import OrderedDict
+import datetime
+from functools import partial
+from functools import reduce
+import io
+import itertools
+import json
+import logging
+import operator
+import os
+import re
+import sys
+import tempfile
+import zipfile
+
+import pkg_resources
+from reportlab.platypus import SimpleDocTemplate
+from reportlab.lib.pagesizes import A4
+
+from transform.transformers.ImageTransformer import ImageTransformer
+from transform.transformers.PDFTransformer import PDFTransformer
+
+__doc__ = """
+python -m transform.transformers.MWSSTransformer \
+< tests/replies/eq-mwss-test-submission.json > test-output.zip
+
+"""
+
+
+class Survey:
+
+    Identifiers = namedtuple("Identifiers", [
+        "batchNr", "seqNr", "ts", "txId", "surveyId", "instId",
+        "userTs", "userId", "ruRef", "ruChk", "period"
+    ])
+
+    @staticmethod
+    def parse_timestamp(text):
+        cls = datetime.datetime
+
+        if text.endswith("Z"):
+            return cls.strptime(text, "%Y-%m-%dT%H:%M:%SZ").replace(
+                tzinfo=datetime.timezone.utc
+            )
+
+        try:
+            return cls.strptime(text, "%Y-%m-%dT%H:%M:%S.%f%z")
+        except ValueError:
+            pass
+
+        try:
+            return cls.strptime(text.partition(".")[0], "%Y-%m-%dT%H:%M:%S")
+        except ValueError:
+            pass
+
+        try:
+            return cls.strptime(text, "%Y-%m-%d").date()
+        except ValueError:
+            pass
+
+        try:
+            return cls.strptime(text, "%d/%m/%Y").date()
+        except ValueError:
+            return None
```

Comment:
ok - might be worth logging at that point then, as that should be the exception that matters.